app/scripts/controllers/overview.js

```diff
@@ -498,6 +538,173 @@ angular.module('openshiftConsole')
+            angular.forEach(map, function(resources, serviceName) {
+              var service = $scope.services[serviceName];
+              if (service) {
+                angular.forEach(resources, function(resource) {
+                  topologyRelations.push({ source: makeId(service), target: makeId(resource) });
+                });
+              }
+            });
+          });
+
+          // A special case, not related to services
+          angular.forEach($scope.podsByDeployment, function(pods, deploymentName) {
+            var deployment = $scope.deploymentsByName[deploymentName];
+            if (deployment && makeId(deployment) in topologyItems) {
+              angular.forEach(pods, function(pod) {
+          topologyItems[makeId(pod)] = pod;
+                topologyRelations.push({ source: makeId(deployment), target: makeId(pod) });
+              });
+            }
+          });
+
+          // Link deployment configs to their deployment
+          angular.forEach($scope.deploymentsByName, function(deployment, deploymentName) {
+            var deploymentConfig, annotations = deployment.metadata.annotations || {};
+            var deploymentConfigName = annotations["openshift.io/deployment-config.name"] || deploymentName;
+            if (deploymentConfigName && $scope.deploymentConfigs) {
+              deploymentConfig = $scope.deploymentConfigs[deploymentConfigName];
+              if (deploymentConfig) {
+                topologyRelations.push({ source: makeId(deploymentConfig), target: makeId(deployment) });
+              }
+            }
+          });
+
+          $scope.$evalAsync(function() {
+            $scope.topologyItems = topologyItems;
+            $scope.topologyRelations = topologyRelations;
+          });
+        }
+
+        function updateTopologyLater() {
+          if (!updateTimeout) {
+            updateTimeout = window.setTimeout(updateTopology, 100);
+          }
+        }
+
+        $scope.$on("select", function(ev, resource) {
+          $scope.$apply(function() {
+            $scope.topologySelection = resource;
+          });
+        }, true);
+
+        $scope.$watch('renderOptions.overviewMode', function(mode) {
+          if (mode === 'topology') {
+            $(window).on('click.topologyItem', function(evt) {
+              var g = $(evt.target).closest('g', $('.kube-topology'));
+              $('g', '.kube-topology').not(g).popover('hide');
+              if (g.length) {
+                if (!(g.data('bs.popover'))) {
+                  g.popover({
+                    container: 'body',
+                    content: function() {
+                      return $filter('sentenceCase')($scope.topologySelection.kind) + ' <a href="' + $filter('navigateResourceURL')($scope.topologySelection) + '">' + $scope.topologySelection.metadata.name + '</a>';
+                    },
+                    html: true,
+                    placement: 'top',
+                    trigger: 'manual'
```

Comment:
We tried that, @benjaminapetersen.  Unfortunately, it didn't work.  :(  I can't remember why exactly...